[Bug]: test_vllm_port.py::test_get_vllm_port_uri  fails with AssertionError: Regex pattern did not match

### Your current environment

Issue seems to have been introduced in https://github.com/vllm-project/vllm/pull/18209

### 🐛 Describe the bug

Test fails with below traceback.

```
    def test_get_vllm_port_uri():
        """Test when VLLM_PORT is set to a URI."""
        with (patch.dict(os.environ, {"VLLM_PORT": "tcp://localhost:5678"},
                         clear=True),
>             pytest.raises(ValueError, match="appears to be a URI")):
E             AssertionError: Regex pattern did not match.
E              Regex: 'appears to be a URI'
E              Input: "VLLM_PORT 'tcp://localhost:5678' must be a valid integer"

tests/test_vllm_port.py:34: AssertionError
```


### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

--- a/.buildkite/test-pipeline.yaml
+++ b/.buildkite/test-pipeline.yaml
@@ -199,8 +199,9 @@ steps:
   - tests/test_sequence
   - tests/test_config
   - tests/test_logger
+  - tests/test_vllm_port
   commands:
-  - pytest -v -s engine test_sequence.py test_config.py test_logger.py
+  - pytest -v -s engine test_sequence.py test_config.py test_logger.py test_vllm_port.py
   # OOM in the CI unless we run this separately
   - pytest -v -s tokenization
 

--- a/vllm/envs.py
+++ b/vllm/envs.py
@@ -158,17 +158,13 @@ def get_vllm_port() -> Optional[int]:
         return int(port)
     except ValueError as err:
         from urllib.parse import urlparse
-        try:
-            parsed = urlparse(port)
-            if parsed.scheme:
-                raise ValueError(
-                    f"VLLM_PORT '{port}' appears to be a URI. "
-                    "This may be caused by a Kubernetes service discovery issue"
-                    "check the warning in: https://docs.vllm.ai/en/stable/usage/env_vars.html"
-                )
-        except Exception:
-            pass
-
+        parsed = urlparse(port)
+        if parsed.scheme:
+            raise ValueError(
+                f"VLLM_PORT '{port}' appears to be a URI. "
+                "This may be caused by a Kubernetes service discovery issue,"
+                "check the warning in: https://docs.vllm.ai/en/stable/serving/env_vars.html"
+            ) from None
         raise ValueError(
             f"VLLM_PORT '{port}' must be a valid integer") from err
 

